Sources/ManageMaintenance.php

```diff
@@ -2104,7 +2104,12 @@ function get_integration_hooks_data($start, $per_page, $sort, $filtered_hooks, $
 	foreach ($filtered_hooks as $hook => $functions)
 		foreach ($functions as $rawFunc)
 		{
-			$hookParsedData = parse_integration_hook($hook, $rawFunc, $files);
+			$hookParsedData = parse_integration_hook($hook, $rawFunc);
+
+			// Handle hooks pointing outside the sources directory.
+			if ($hookParsedData['absPath'] != '' && (!isset($files[$hookParsedData['absPath']]) || file_exists($hookParsedData['absPath'])))
```

Comment:
```suggestion
			if ($hookParsedData['absPath'] != '' && !isset($files[$hookParsedData['absPath']]) && file_exists($hookParsedData['absPath']))
```
 This should avoid running files through `get_defined_functions_in_file()` again if we already did them above.